### PR TITLE
Adds Mortal Spin and Population Bomb

### LINKED ISSUE #2493: MOVE_POPULATION_BOMB Effect

### LINKED ISSUE #2497: MOVE_MORTAL_SPIN Effect

### PR DESCRIPTION
**Pull request by AlexOn1ine**
Closes #2497 (Credit: Lunos)
Closes #2493

### CODE REVIEW

**Review by DizzyEggg — CHANGES_REQUESTED** [review]
There were some changes to Triple Kick and multi hit moves in general. I would feel more comfortable merging this PR if we got tests for Triple Kick and moves like Double Kick, Fury Swipes, etc. Also their interactions with Skill Link, when they can miss, when the can't miss, etc.

**Comment by AlexOn1ine** [comment]
Added Tests for Skill Link for `EFFECT_MULTI_HIT` and some tiny logic for Loaded Dice.

I wasn't able to write tests for Triple Kick and Population Bomb because it doesn't seem you can check their accuracy correctly so @AsparagusEduardo suggested to just add the `TO_DO_BATTLE_TEST`

File: src/battle_script_commands.c [hunk 1]
```diff
@@ -1722,10 +1722,11 @@ static void Cmd_accuracycheck(void)
             gBattlescriptCurrInstr = cmd->nextInstr;
     }
     else if (gSpecialStatuses[gBattlerAttacker].parentalBondState == PARENTAL_BOND_2ND_HIT
-        || (gSpecialStatuses[gBattlerAttacker].multiHitOn && (gBattleMoves[move].effect != EFFECT_TRIPLE_KICK
-        || GetBattlerAbility(gBattlerAttacker) == ABILITY_SKILL_LINK)))
+        || (gSpecialStatuses[gBattlerAttacker].multiHitOn
+        && (GetBattlerAbility(gBattlerAttacker) == ABILITY_SKILL_LINK || GetBattlerHoldEffect(gBattlerAttacker, TRUE)
+        || !(gBattleMoves[move].effect == EFFECT_TRIPLE_KICK || gBattleMoves[move].effect == EFFECT_POPULATION_BOMB))))
```
[the remark this answers is not in the document]
Comment (AlexOn1ine) [reply src/battle_script_commands.c]: Fixed.

Also added a small test for Population Bomb, not sure how useful it is but at least it tests that 10 Hits are possible.

**Comment by AlexOn1ine** [comment]
The tests on github don't pass 
```

- Tests FAILED :       2    Add TESTS='X' to run tests with the defined prefix.
  - Sand Veil increases evasion during sandstorm (1/?).
  - Snow Cloak increases evasion during hail (1/?).
  ```
  but they pass for me 🤔
```
[0] Sand Veil prevents damage from sandstorm: PASS
[0] Sand Veil increases evasion during sandstorm: PASS
- Tests PASSED:        2
- Tests TOTAL:         2
- ```

**Comment by mrgriffin** [comment]
> The tests on github don't pass
> 
> ```
> 
> - Tests FAILED :       2    Add TESTS='X' to run tests with the defined prefix.
>   - Sand Veil increases evasion during sandstorm (1/?).
>   - Snow Cloak increases evasion during hail (1/?).
> ```
> 
> but they pass for me thinking
> 
> ```
> [0] Sand Veil prevents damage from sandstorm: PASS
> [0] Sand Veil increases evasion during sandstorm: PASS
> - Tests PASSED:        2
> - Tests TOTAL:         2
> - ```
> ```

If you run with `make check -j2` do they fail for you? (Yay #3132)

**Comment by AlexOn1ine** [comment]
> > The tests on github don't pass
> > ```
> > 
> > - Tests FAILED :       2    Add TESTS='X' to run tests with the defined prefix.
> >   - Sand Veil increases evasion during sandstorm (1/?).
> >   - Snow Cloak increases evasion during hail (1/?).
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > but they pass for me thinking
> > ```
> > [0] Sand Veil prevents damage from sandstorm: PASS
> > [0] Sand Veil increases evasion during sandstorm: PASS
> > - Tests PASSED:        2
> > - Tests TOTAL:         2
> > - ```
> > ```
> 
> If you run with `make check -j2` do they fail for you? (Yay #3132) You should get the same output as [on CI](https://github.com/rh-hideout/pokeemerald-expansion/pull/3178/checks). (Including the `[0]` and `[1]` things)

All good
```
- Tests PASSED:        480
- Tests KNOWN_FAILING: 11
- Tests TO_DO:         40
- Tests TOTAL:         531
```

File: data/battle_scripts_1.s [hunk 2]
```diff
@@ -435,6 +435,17 @@ gBattleScriptsForMoveEffects::
 	.4byte BattleScript_EffectHit                     @ EFFECT_COLLISION_COURSE
 	.4byte BattleScript_EffectSpinOut                 @ EFFECT_SPIN_OUT
 	.4byte BattleScript_EffectMakeItRain              @ EFFECT_MAKE_IT_RAIN
+	.4byte BattleScript_EffectHit                     @ EFFECT_POPULATION_BOMB
+	.4byte BattleScript_EffectMortalSpin              @ EFFECT_MORTAL_SPIN
+
+BattleScript_EffectMortalSpin:
+	call BattleScript_EffectHit_Ret
+	rapidspinfree
```
Comment (Skeli789) [inline data/battle_scripts_1.s:443]: This is not completely correct. Mortal Spin only clears hazards if the user hasn't fainted (eg. due to the effects like Rough Skin).

Comment (AlexOn1ine) [reply data/battle_scripts_1.s]: Should be fixed

**Review by DizzyEggg — APPROVED** [review]

**Pull request merged**